app/views/hyrax/my/_admin_set_action_menu.html.erb

```diff
@@ -18,20 +18,32 @@
                     class: 'itemicon itemedit',
                     title: t("hyrax.dashboard.my.action.edit_admin_set") do %>
           <%= t("hyrax.dashboard.my.action.edit_admin_set") %>
+        <% end %>         
+      </li>
+    <% else %>
+      <li role="menuitem" tabindex="-1">
+        <%= link_to "#",
+            class: 'itemicon itemtrash edit-collection-deny-button',
+            title: t("hyrax.dashboard.my.action.edit_collection") do %>
+          <%= t("hyrax.dashboard.my.action.edit_collection") %>
+        <% end %>
+      </li>
+    <% end %>
+    <% unless admin_set_presenter.disable_delete? %>
+      <li role="menuitem" tabindex="-1">
+        <%= link_to hyrax.admin_admin_set_path(id),
+                      class: 'itemicon itemtrash delete-collection-button',
+                      title: t("hyrax.dashboard.my.action.delete_admin_set"),       
+                      data: { totalitems: admin_set_presenter.total_items , 
+                              membership: admin_set_presenter.collection_type_is_require_membership? ,
+                              hasaccess: (can? :edit, admin_set_presenter.solr_document) } do %>
+          <%= t("hyrax.dashboard.my.action.delete_admin_set") %>
         <% end %>
       </li>
-      <% unless admin_set_presenter.disable_delete? %>
-        <li role="menuitem" tabindex="-1">
-          <%= link_to hyrax.admin_admin_set_path(id),
-                      class: 'itemicon itemtrash',
-                      title: t("hyrax.dashboard.my.action.delete_admin_set"),
-                      method: :delete,
-                      data: {
-                        confirm: t("hyrax.dashboard.my.action.admin_set_confirmation", application_name: application_name) } do %>
-            <%= t("hyrax.dashboard.my.action.delete_admin_set") %>
-          <% end %>
-        </li>
-      <% end %>
     <% end %>
   </ul>
 </div>
+<!-- Render delete modal -->
+<%= render 'modal_delete_empty_admin_set', id: id %>
```

Comment:
What is the reasoning behind rendering these modals here, but others in _list_collections?